autorest/error.go

```diff
@@ -28,6 +28,9 @@ type DetailedError struct {
 
 	// Message is the error message.
 	Message string
+
+	// Service Error is the response body of failed API.
+	ServiceError interface{}
```

Comment:
@jkapil there's a discrepancy with the type and the comment. If you are unmarshaling the error into this object, it's perfectly fine; just don't call it "the response body".